### DISCUSSION TITLE
能否在markdown里读取环境变量？

### DISCUSSION THREAD

**Discussion by alienzhangyw** [umijs/dumi#1985] [Q&A] [answered]
我想通过设置自定义环境变量的方式，全局替换多个md文件中的一些文本，这个要怎么实现？
另外.dumirc里面能否读取环境变量？

**Comment by PeachScript** [comment]
> 通过设置自定义环境变量的方式，全局替换多个md文件中的一些文本

不支持，md 不是 js

> .dumirc里面能否读取环境变量

可以，配置文件会在 Node.js 运行时内执行

**Reply by alienzhangyw** [reply]
不能像webpack一样在编译成html时进行替换吗？
我有一个实现思路，不知道可不可行。因为单个md是可以定义变量的，比如`[key]: value`，能不能实现在md的头部配置中设置某些变量的值是process.env里面的键，比如这样：
```markdown
---
title: Some document
group:
  title: Some group
  order: 0
order: 1
variables:
  variableA: process.env.DUMI_CUSTOM_VARIABLE_A  // 假设是abc
  variableB: process.env.DUMI_CUSTOM_VARIABLE_B  // 假设是baidu.com
---
```
然后编译时自动把变量定义加到md正文前面(或者文件末尾)：
```
[variableA]: abc
[variableB]: baidu.com
```

如果这样可以的话，也可以直接在.dumirc.ts里定义md变量，编译时自动加到每一个md文件里。

**Reply by Wxh16144** [reply, marked as answer]
https://github.com/umijs/dumi/issues/2008#issuecomment-2095623168 已经失去一位开发者用户 2333~

可以考虑参考 docusaurus 写一个 dumi 插件，完善一下生态 （不过感觉这个需求有些太小众了

**Reply by alienzhangyw** [reply]
主要还是因为docusaurus能写openapi文档，而且支持mdx才改用的
